python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -225,81 +236,35 @@ def unlock_transaction(self, this_id):
         self.hard_lock[this_id].release()
 
     # Methods intended to be called from ''outside code''
+    @synchronised
     def __getitem__(self, this_id):
         """ Returns the Ganga Object with the given id.
             Raise RegistryKeyError"""
         logger.debug("__getitem__")
-        #self._lock.acquire()
         try:
-            self.lock_transaction( this_id, "_getitem")
-
-            real_id = None
-            if type(this_id) is int:
-                if this_id >= 0:
-                    ## +ve integer, should be in dictionary
-                    real_id = this_id
-                    this_obj = self._objects[this_id]
-                else:
-                    ## -ve integer should be a relative object in dictionary
-                    real_id = self._objects.keys()[this_id]
-                    this_obj = self._objects[real_id]
-            else:
-                ## NOT an integer, maybe it's a slice or other?
-                this_obj = self._objects[this_id]
-
-            logger.debug("found_object")
-
-            found_id = None
-            if hasattr(this_obj, _id_str):
-                found_id = getattr(this_obj, _id_str)
-            if hasattr(this_obj, _reg_id_str):
-                found_id = getattr(this_obj, _reg_id_str)
-            if found_id is not None and real_id is not None:
-                assert( found_id == real_id )
```

Comment:
@drmarkwslater I don't see where the object is locked here. The Registry is locked by the decorator from what I'm reading and we have a RLock which is in the ```__get__``` method but these shouldn't be interfering as the Registry doesn't inherit from GangaObject.

I think all that is happening is that within a registry lock we're calling for an attribute of a GangaObject which should manage it's own (separate) (R)locks.

If the ganga object is taking control of locking/unlocking the repo then I think this should be addressed as I don't think it's the place of the GangaObject to lock/unlock the repo at all.

If I've misunderstood something I'm keen to be corrected as this is all a bit complex.